Cpp2IL.Core/Utils/X64CallingConventionResolver.cs

```diff
@@ -0,0 +1,306 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Cpp2IL.Core.ISIL;
+using Cpp2IL.Core.Model.Contexts;
+using LibCpp2IL.BinaryStructures;
+using LibCpp2IL.PE;
+
+namespace Cpp2IL.Core.Utils;
+
+#pragma warning disable IDE0305, IDE0300
+
+public static class X64CallingConventionResolver
+{
+    // TODO: GCC(Linux) ABI
+
+    // This class must be used in good faith.
+    // If that's not possible, uncomment the binary type checks.
+    // This *will* break everything on x32.
+
+    const int ptrSize = 8;

```

Comment:
Also available as:  **AppContext.Binary.PointerSize**

Reply:
I know. X64 explicitly states the pointer size is going to be 8 bytes, so I hardcoded it as a constant to avoid unnecessary property gets and because declaring ptrSize outside of the resolution method would look better than if it was inside of it.